src/Data/Nat/Binary/Minus.agda

```diff
@@ -0,0 +1,364 @@
+  2ᵇ + (double x∸y + double y)         ≡⟨ cong (2ᵇ +_) (sym (double-distrib-+ x∸y y)) ⟩
+  2ᵇ + double (x∸y + y)                ≡⟨ cong ((2ᵇ +_) ∘ double) $ sym $
+                                         x≥y⇒x≡[x∸y]+y y≤x ⟩
+  double 1ᵇ + double x                 ≡⟨ sym (double-distrib-+ 1ᵇ x) ⟩
+  double (1ᵇ + x)                      ≡⟨ cong double (sym (suc≗1+ x)) ⟩
+  double (suc x)                       ≡⟨ sym (2[1+_]-double-suc x) ⟩
+  2[1+ x ]                             ∎
+  where
+  open ≤-Reasoning
+  x∸y = x ∸ y
+  eq = let _+_ = _⊕_;  a = var (# 0);  b = var (# 1);  c = var (# 2)
+       in
+       prove 3 ((a + b) + (a + c)) ((a + a) + (b + c)) (1ᵇ ∷ (double x∸y) ∷ (double y) ∷ [])
+x≥y⇒x≡[x∸y]+y {1+[2 x ]}      {zero}          (inj₁ 0<odd)          =  refl
+x≥y⇒x≡[x∸y]+y {x'@(1+[2 x ])} {y'@(2[1+ y ])} (inj₁ (even<odd y<x)) =  sym $ begin-equality
+  (x' ∸ y') + y'                        ≡⟨ cong₂ _+_ (odd∸even-for> y<x)
+                                                     (2[1+_]-double-suc y) ⟩
+  pred (double x∸y) + (double (suc y))  ≡⟨ pred[x]+y≡x+pred[y] 2[x∸y]≢0 2[suc[y]]≢0 ⟩
+  2[x∸y] + pred (double (suc y))        ≡⟨ cong ((2[x∸y] +_) ∘ pred) (double-suc y) ⟩
+  2[x∸y] + pred (suc 1ᵇ + 2y)           ≡⟨ cong ((2[x∸y] +_) ∘ pred) (suc-+ 1ᵇ 2y) ⟩
+  2[x∸y] + pred (suc (1ᵇ + 2y))         ≡⟨ cong (2[x∸y] +_) (pred-suc (1ᵇ + 2y)) ⟩
+  2[x∸y] + (1ᵇ + 2y)                    ≡⟨ x∙yz≈y∙xz 2[x∸y] 1ᵇ 2y ⟩
+  1ᵇ + (2[x∸y] + 2y)                    ≡⟨ cong (1ᵇ +_) $ sym $ double-distrib-+ x∸y y ⟩
+  1ᵇ + double (x∸y + y)                 ≡⟨ cong ((1ᵇ +_) ∘ double) $ sym $
+                                           x≥y⇒x≡[x∸y]+y (inj₁ y<x) ⟩
+  1ᵇ + 2x                               ≡⟨ sym (suc≗1+ 2x) ⟩
+  suc 2x                                ≡⟨ sym (1+[2_]-suc-double x) ⟩
+  x'                                    ∎
+  where
+  open ≤-Reasoning
+  x∸y = x ∸ y;  2x = double x;  2y = double y;  2[x∸y] = double x∸y;  x∸y>0 = x<y⇒0<y∸x y<x
+  x∸y≢0       = >⇒≢ x∸y>0
+  2[x∸y]≢0    = x≢0⇒double[x]≢0 x∸y≢0
+  2[suc[y]]≢0 = x≢0⇒double[x]≢0 (suc≢0 {y})
+x≥y⇒x≡[x∸y]+y {x'@(1+[2 x ])} {y'@(1+[2 y ])} (inj₁ (odd<odd y<x)) =  sym $ begin-equality
+  (x' ∸ y') + y'                 ≡⟨ cong ((x' ∸ y') +_) (1+[2_]-suc-double y) ⟩
+  double x∸y + (suc (double y))  ≡⟨ x+suc[y]≡suc[x]+y (double x∸y) (double y) ⟩
+  suc (double x∸y) + double y    ≡⟨ suc-+ (double x∸y) (double y) ⟩
+  suc (double x∸y + double y)    ≡⟨ cong suc $ sym $ double-distrib-+ x∸y y ⟩
+  suc (double (x∸y + y))         ≡⟨ sym (1+[2_]-suc-double _) ⟩
+  1+[2 (x∸y + y) ]               ≡⟨ cong 1+[2_] $ sym $ x≥y⇒x≡[x∸y]+y (inj₁ y<x) ⟩
+  1+[2 x ]                       ∎
+  where open ≤-Reasoning;  x∸y = x ∸ y
+x≥y⇒x≡[x∸y]+y {x'@(1+[2 x ])} {1+[2 .x ]} (inj₂ refl) =  sym $ begin-equality
+  (x' ∸ x') + x'  ≡⟨ cong (_+ x') (x∸x≡0 x') ⟩
+  0ᵇ + x'         ≡⟨ sym (+-identityˡ x') ⟩
+  x'              ∎
+  where open ≤-Reasoning
+
+[x+y]∸y≡x :  ∀ x y → (x + y) ∸ y ≡ x
```

Comment:
Please could you stop adding the double spaces after the colons and equals signs. You only do so inconsistently. I know it allows aligning in literate Agda but the standard library doesn't tend to be used for literate Agda.

Reply:
I did this not due to literate Agda, but in order to make the source more readable.                                                                    
All right, I remove these doubles.